charts/k8s-service/templates/_deployment_spec.tpl

```diff
@@ -303,6 +303,13 @@ spec:
     {{- end }}
     {{- /* END IMAGE PULL SECRETS LOGIC */ -}}
 
+    {{- /* START TERMINATION GRACE PERIOD LOGIC */ -}}
+    {{- if .Values.terminationGracePeriodSeconds }}    
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds | 30 }}
```

Comment:
I think the `| 30` is never executed, since this block is only rendered if `.Values.terminationGracePeriodSeconds` is truthy (because of the `if`). It's better to rely on the defaults of the k8s API anyway instead of rendering an explicit value, so we should just remove the `| 30` component here:

```suggestion
      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
```